src/mm/page_alloc.rs

```diff
@@ -9,4 +10,28 @@ pub enum AllocatorError {
 pub trait PageAllocator {
     fn alloc_pages(&mut self, page_count: usize) -> Result<PAddr, AllocatorError>;
     fn dealloc_pages(&mut self, ptr: PAddr) -> Result<(), AllocatorError>;
+
+    fn page_size(&self) -> usize;
+}
+
+static mut GLOBAL_ALLOCATOR: Option<FirstFitPageAllocator> = None;
```

Comment:
I wonder if it would make sense to instead have a Mutex here for safer operations. We could probably have less unsafe in the code below, but I don't think it would change much. I'm fine with this `Option<FirstFitPageAllocator>`

Reply:
Fixed by 7614aec